orchestrator/test_runner/src/happy_path_v2.rs

```diff
@@ -64,34 +64,41 @@ pub async fn happy_path_test_v2(
         "Successfully deployed new ERC20 representing FooToken on Cosmos with event nonce {}",
         ending_event_nonce
     );
-
-    let start = Instant::now();
     // the erc20 representing the cosmos asset on Ethereum
     let mut erc20_contract = None;
-    while Instant::now() - start < TOTAL_TIMEOUT {
-        let res = grpc_client
-            .denom_to_erc20(DenomToErc20Request {
-                denom: token_to_send_to_eth.clone(),
-            })
-            .await;
-        if let Ok(res) = res {
-            let erc20 = res.into_inner().erc20;
+    match tokio::time::timeout(TOTAL_TIMEOUT, async {
+        loop {
+            let res = grpc_client
+                .denom_to_erc20(DenomToErc20Request {
+                    denom: token_to_send_to_eth.clone(),
+                })
+                .await;
+
+            if let Ok(res) = res {
+                let erc20 = res.into_inner().erc20;
+                erc20_contract = Some(erc20);
+                break;
+            }
+            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+        }
+        
+    }).await {
+        Ok(_) => {
```

Comment:
Should we be capturing `erc20` here to include it in the `info!` message like in the original version?

Reply:
Fixed this with this commit: https://github.com/PeggyJV/gravity-bridge/pull/274/commits/14ddc5ce7005ce4814ef4041428b0eea452a7849

What do you think about [replacing](https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html) `match` with `if let`?

```
if let Ok(res) = tokio::time::timeout(TOTAL_TIMEOUT, async {
            loop {
                let res = grpc_client
                    .denom_to_erc20(DenomToErc20Request {
                        denom: token_to_send_to_eth.clone(),
                    })
                    .await;

                if let Ok(res) = res {
                    break res;
                }
                tokio::time::sleep(std::time::Duration::from_secs(1)).await;
            }
        })
        .await {
        let erc20 = res.into_inner().erc20;
        erc20_contract = Some(erc20);
        info!(
            "Successfully adopted {} token contract",
            token_to_send_to_eth
        );
    } else {
                panic!(
                    "Cosmos did not adopt the ERC20 contract for {} it must be invalid in some way",
                    token_to_send_to_eth
                );
            }
```
I'll still leave this open for discussion

Reply:
I think either works, but the `match` version feels easier to read. Just a subjective opinion though.

Reply:
ok